qiskit/optimization/problems/variables.py

```diff
@@ -260,14 +259,18 @@ def add(self, obj=None, lb=None, ub=None, types="", names=None, columns=None):
             raise QiskitOptimizationError(
                 "Please use LinearConstraintInterface instead of columns.")
 
+        start = self.get_num()
         arg_list = init_list_args(lb, ub, types, names)
         arg_lengths = [len(x) for x in arg_list]
         if len(arg_lengths) == 0:
-            return range(0)
+            return range(start, start)
         max_length = max(arg_lengths)
         for arg_length in arg_lengths:
             if arg_length > 0 and arg_length != max_length:
                 raise QiskitOptimizationError("inconsistent arguments")
+        if max_length == 0:
+            return range(start, start)
+        assert max_length > 0
```

Comment:
This `assert` is never reached and be dropped.

Reply:
Done